crates/core/src/node/inner/in_memory_inner.rs

```diff
@@ -471,10 +473,22 @@ impl InMemoryNodeInner {
         } = self.run_l2_tx_raw(l2_tx.clone(), vm)?;
 
         if let ExecutionResult::Halt { reason } = result.result {
-            // Halt means that something went really bad with the transaction execution (in most cases invalid signature,
-            // but it could also be bootloader panic etc).
-            // In such case, we should not persist the VM data, and we should pretend that transaction never existed.
-            anyhow::bail!("Transaction HALT: {reason}");
+            let reason_clone = reason.clone();
+
+            let handle = tokio::runtime::Handle::current();
+            let halt_error =
+                std::thread::spawn(move || handle.block_on(reason_clone.to_halt_error()))
+                    .join()
+                    .expect("Thread panicked");
```

Comment:
Do we need it though? Given that this function is being invoked (transitively) from `async` function already, can't we just make the function itself async? The `std::thread::spawn` hack is normally used when you can't make the function async (e.g. when it's a trait implementation) and it does have some performance penalty.